src/main/webapp/app/shared/util/firebase/firebase-level-utils.tsx

```diff
@@ -0,0 +1,113 @@
+import { DX_LEVELS, PX_LEVELS, TX_LEVELS } from 'app/shared/model/firebase/firebase.model';
+import { SortOrder, sortByIndex } from './firebase-utils';
+import {
+  ALL_LEVELS,
+  DIAGNOSTIC_LEVELS_ORDERING,
+  FDA_LEVELS_ORDERING,
+  FDA_LEVEL_KEYS,
+  FDA_LEVEL_KEYS_MAPPING,
+  PROGNOSTIC_LEVELS_ORDERING,
+  THERAPEUTIC_LEVELS_ORDERING,
+  THERAPEUTIC_RESISTANCE_LEVELS,
+  THERAPEUTIC_SENSITIVE_LEVELS,
+} from 'app/config/constants/firebase';
+import React from 'react';
+import { LevelWithDescription } from 'app/shared/text/LevelWithDescription';
+import { PropagationType } from 'app/pages/curation/collapsible/TherapyDropdownGroup';
+import { RealtimeDropdownOptions } from 'app/shared/firebase/input/RealtimeDropdownInput';
+
+export enum LevelType {
+  ALL,
+  THERAPEUTIC,
+  THERAPEUTIC_SENSITIVE,
+  THERAPEUTIC_RESISTANCE,
+  PROGNOSTIC,
+  DIAGNOSTIC,
+  FDA,
+}
+
+export type LEVELS = TX_LEVELS | PX_LEVELS | DX_LEVELS | FDA_LEVEL_KEYS;
```

Comment:
Why no keep the consistency use `FDA_LEVELS`?

Reply:
I couldn't use FDA_LEVELS because it clashes with the value from the TX_LEVELS enum. For instance, `FDA_LEVELS.LEVEL_FDA1 = '1'` and `TX_LEVELS.LEVEL_1 = '1'`. I created a mapping between `FDA_LEVEL_KEYS` with `FDA_LEVELS` in constants file